tools/accuracy_checker/accuracy_checker/adapters/detection.py

```diff
@@ -705,3 +705,148 @@ def _find_output(predictions):
+                        dense_cx = [x * steps[k] / image_size[1] for x in [j + 0, j + 0.5]]
+                        dense_cy = [y * steps[k] / image_size[0] for y in [i + 0, i + 0.5]]
+                        for cy, cx in itertools.product(dense_cy, dense_cx):
+                            anchors.append([cx, cy, s_kx, s_ky])
+                    else:
+                        cx = (j + 0.5) * steps[k] / image_size[1]
+                        cy = (i + 0.5) * steps[k] / image_size[0]
+                        anchors.append([cx, cy, s_kx, s_ky])
+        anchors = np.clip(anchors, 0, 1)
+
+        return anchors
+
+    def process(self, raw, identifiers=None, frame_meta=None):
+        """
+        Args:
+            identifiers: list of input data identifiers
+            raw: output of model
+        Returns:
+            list of DetectionPrediction objects
+        """
+
+        raw_outputs = self._extract_predictions(raw, frame_meta)
+
+        batch_scores = raw_outputs[self.scores_out]
+        batch_boxes = raw_outputs[self.boxes_out]
+
+        result = []
+        for identifier, scores, boxes, meta in zip(identifiers, batch_scores, batch_boxes, frame_meta):
+            detections = {'labels': [], 'scores': [], 'x_mins': [], 'y_mins': [], 'x_maxs': [], 'y_maxs': []}
+            image_info = meta.get("image_info")[0:2]
+
+            # Prior boxes
+            feature_maps = [[math.ceil(image_info[0] / step), math.ceil(image_info[1] / step)] for step in
+                            self.steps]
+            prior_data = self.prior_boxes(feature_maps, self.min_sizes, image_info, self.steps)
+
+             # Boxes
+            boxes[:, :2] = self.variance[0] * boxes[:, :2]
+            boxes[:, 2:] = self.variance[1] * boxes[:, 2:]
+            boxes[:, :2] = boxes[:, :2] * prior_data[:, 2:] + prior_data[:, :2]
+            boxes[:, 2:] = np.exp(boxes[:, 2:]) * prior_data[:, 2:]
+
+            for label, score in enumerate(np.transpose(scores)):
+
+                # Skip background label
+                if label == 0:
+                    continue
+
+                # Filter out detections with score < confidence_threshold
+                mask = score > self.confidence_threshold
+                filtered_boxes, filtered_score = boxes[mask, :], score[mask]
+                if filtered_score.size == 0:
+                    continue
+
+                # Transform to format (x_min, y_min, x_max, y_max)
+                x_mins = (filtered_boxes[:, 0] - 0.5 * filtered_boxes[:, 2])
+                y_mins = (filtered_boxes[:, 1] - 0.5 * filtered_boxes[:, 3])
+                x_maxs = (filtered_boxes[:, 0] + 0.5 * filtered_boxes[:, 2])
+                y_maxs = (filtered_boxes[:, 1] + 0.5 * filtered_boxes[:, 3])
+
+                # Apply NMS
+                keep = NMS.nms(x_mins, y_mins, x_maxs, y_maxs, filtered_score, self.nms_threshold,
+                               include_boundaries=False, keep_top_k=self.keep_top_k)
```

Comment:
in pytorch implementation, there are 2 top_k coefficients (before NMS [`top_k`](https://github.com/zisianw/FaceBoxes.PyTorch/blob/master/test.py#L137) and after [`keep_top_k`](https://github.com/zisianw/FaceBoxes.PyTorch/blob/master/test.py#L148))
I think you need more carefully look on their usage. It can have influence on accuracy

Reply:
I checked their default coefficients(top_k =5000, keep_top_k = 750). map decreased to 83.51 from 83.57
Optimal result(map 83.57) is achieved when top_k =750, keep_top_k = 750. When keep_top_k decreases  optimal result does not change